tests/pytests/unit/utils/jinja/test_custom_extensions.py

```diff
@@ -0,0 +1,1088 @@
+    assert rendered == "barred, it works"
+
+    source = (
+        '{{ bar }}, {% load_json as docu %}{"foo": "it works", "{{ bar }}":'
+        ' "baz"}{% endload %}' + "{{ docu.foo }}"
+    )
+
+    rendered = env.from_string(source).render(bar="barred")
+    assert rendered == "barred, it works"
+
+    with pytest.raises(exceptions.TemplateSyntaxError):
+        env.from_string(
+            "{% load_yamle as document %}{foo, bar: it works}{% endload %}"
+        ).render()
+
+    with pytest.raises(exceptions.TemplateRuntimeError):
+        env.from_string(
+            "{% load_json as document %}{foo, bar: it works}{% endload %}"
+        ).render()
+
+
+def test_load_json():
+    env = Environment(extensions=[SerializerExtension])
+    rendered = env.from_string(
+        '{% set document = \'{"foo": "it works"}\'|load_json %}{{ document.foo }}'
+    ).render()
+    assert rendered == "it works"
+
+    rendered = env.from_string(
+        "{% set document = document|load_json %}{{ document.foo }}"
+    ).render(document='{"foo": "it works"}')
+    assert rendered == "it works"
+
+    # bad quotes
+    with pytest.raises(exceptions.TemplateRuntimeError):
+        env.from_string("{{ document|load_json }}").render(
+            document="{'foo': 'it works'}"
+        )
+
+    # not a string
+    with pytest.raises(exceptions.TemplateRuntimeError):
+        env.from_string("{{ document|load_json }}").render(document={"foo": "it works"})
+
+
+def test_load_yaml_template():
+    loader = DictLoader({"foo": '{bar: "my god is blue", foo: [1, 2, 3]}'})
+    env = Environment(extensions=[SerializerExtension], loader=loader)
+    rendered = env.from_string('{% import_yaml "foo" as doc %}{{ doc.bar }}').render()
+    assert rendered == "my god is blue"
+
+    with pytest.raises(exceptions.TemplateNotFound):
+        env.from_string('{% import_yaml "does not exists" as doc %}').render()
+
+
+def test_load_json_template():
+    loader = DictLoader({"foo": '{"bar": "my god is blue", "foo": [1, 2, 3]}'})
+    env = Environment(extensions=[SerializerExtension], loader=loader)
+    rendered = env.from_string('{% import_json "foo" as doc %}{{ doc.bar }}').render()
+    assert rendered == "my god is blue"
+
+    with pytest.raises(exceptions.TemplateNotFound):
+        env.from_string('{% import_json "does not exists" as doc %}').render()
+
+
+def test_load_text_template():
+    loader = DictLoader({"foo": "Foo!"})
+    env = Environment(extensions=[SerializerExtension], loader=loader)
+
+    rendered = env.from_string('{% import_text "foo" as doc %}{{ doc }}').render()
+    assert rendered == "Foo!"
+
+    with pytest.raises(exceptions.TemplateNotFound):
+        env.from_string('{% import_text "does not exists" as doc %}').render()
+
+
+def test_profile():
+    env = Environment(extensions=[SerializerExtension])
+
+    source = (
+        "{%- profile as 'profile test' %}"
+        + "{% set var = 'val' %}"
```

Comment:
```suggestion
        "{% set var = 'val' %}"
```
The `+` is not really necessary, python will concatenate the strings, right?